### PR TITLE
Add possibility to inject custom telemetry attributes via ConfigMap

### PR DESCRIPTION
**Pull request by jnummelin**
Signed-off-by: Jussi Nummelin <jnummelin@mirantis.com>

## Description

This allows us to add certain "pre-defined" data e.g. from Lens Desktop use case so that we can better identify these uses of k0s in telemetry data.

So essentially, if a ConfigMap `kube-system/k0s-telemetry` exists, we will include all KV pairs to telemetry data.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/telemetry/telemetry.go [hunk 1]
```diff
@@ -159,6 +160,17 @@ func (c Component) sendTelemetry(ctx context.Context) {
...
 }
 
+func (c Component) addCustomData(ctx context.Context, analyticCtx *analytics.Context) {
+	cm, err := c.kubernetesClient.CoreV1().ConfigMaps("kube-system").Get(ctx, "k0s-telemetry", metav1.GetOptions{})
+	if err != nil {
+		c.log.WithError(err).Info("can't get k0s-telemetry configmap, not adding custom data")
+		return
+	}
+	for k, v := range cm.Data {
+		analyticCtx.Extra[k] = v
```
Comment (kke) [inline pkg/telemetry/telemetry.go:170]: As the "extra" is a fake field and its contents are just serialized in the json, this makes it possible to overwrite existing data in the analytics json, accidentally or, maybe in some far fetched scenario, on purpose.

Maybe prefix the keys from the custom data with something like `custom.`?

File: pkg/telemetry/telemetry.go [hunk 2]
```diff
@@ -159,6 +160,17 @@ func (c Component) sendTelemetry(ctx context.Context) {
 	}
 }
 
+func (c Component) addCustomData(ctx context.Context, analyticCtx *analytics.Context) {
+	cm, err := c.kubernetesClient.CoreV1().ConfigMaps("kube-system").Get(ctx, "k0s-telemetry", metav1.GetOptions{})
+	if err != nil {
+		c.log.WithError(err).Info("can't get k0s-telemetry configmap, not adding custom data")
```
Comment (kke) [inline pkg/telemetry/telemetry.go:166]: Constantly seeing this error in the logs may bother users

File: pkg/telemetry/telemetry.go (hunk 1 above)
Comment (kke) [reply pkg/telemetry/telemetry.go:170]: Hmm, I wonder if Extra could take the cm.Data as is, something like `analyticsCtx.Extra["custom"] = cm.Data`

Comment (jnummelin) [reply pkg/telemetry/telemetry.go:170]: > Maybe prefix the keys from the custom data with something like custom.?

This sounds like a good idea to me 👍

**Review by kke — APPROVED** [review]

**Pull request merged**